x-pack/plugins/ml/public/embeddables/single_metric_viewer/single_metric_viewer_embeddable_factory.tsx

```diff
@@ -0,0 +1,350 @@
+            onDestroy();
+            subscriptions.unsubscribe();
+          });
+
+          useReactEmbeddableExecutionContext(
+            services[0].executionContext,
+            // TODO https://github.com/elastic/kibana/issues/180055
+            // @ts-ignore
+            parentApi?.executionContext?.value ?? { name: 'dashboard' },
+            ANOMALY_SINGLE_METRIC_VIEWER_EMBEDDABLE_TYPE,
+            uuid
+          );
+
+          const selectedJobId = singleMetricViewerData?.jobIds[0];
+          // Need to make sure we fall back to `undefined` if `functionDescription` is an empty string,
+          // otherwise anomaly table data will not be loaded.
+          const functionDescription =
+            (singleMetricViewerData?.functionDescription ?? '') === ''
+              ? undefined
+              : singleMetricViewerData.functionDescription;
+          const previousRefresh = usePrevious(lastRefresh ?? 0);
+
+          // Holds the container height for previously fetched data
+          const containerHeightRef = useRef<number>();
+
+          useEffect(function setUpJobsLoaded() {
+            async function loadJobs() {
+              try {
+                await mlJobService.loadJobsWrapper();
+                setJobsLoaded(true);
+              } catch (e) {
+                const errorMessage = extractErrorMessage(e);
+                setError(errorMessage);
+              }
+            }
+            loadJobs();
+            // eslint-disable-next-line react-hooks/exhaustive-deps
+          }, []);
+
+          useEffect(
+            function setUpSelectedJob() {
+              async function fetchSelectedJob() {
+                if (mlApiServices && selectedJobId !== undefined && error === undefined) {
+                  try {
+                    const { jobs } = await mlApiServices.getJobs({ jobId: selectedJobId });
+                    const job = jobs[0];
+                    setSelectedJob(job);
+                  } catch (e) {
+                    const errorMessage = extractErrorMessage(e);
+                    setError(errorMessage);
+                  }
+                }
+              }
+              fetchSelectedJob();
+            },
+            [selectedJobId, mlApiServices, error]
+          );
+
+          useEffect(
+            function setUpAutoZoom() {
+              let zoomDuration: number | undefined;
+              if (selectedJobId !== undefined && selectedJob !== undefined) {
+                zoomDuration = mlTimeSeriesExplorerService?.getAutoZoomDuration(selectedJob);
+                setAutoZoomDuration(zoomDuration);
+              }
+            },
+            // eslint-disable-next-line react-hooks/exhaustive-deps
+            [selectedJobId, selectedJob?.job_id, mlTimeSeriesExplorerService]
+          );
+
+          useEffect(function onUnmount() {
+            return () => {
+              subscriptions.unsubscribe();
+            };
+          }, []);
```

Comment:
```suggestion
```
you already have it in `useUnmount` effect

Reply:
Missed in the refactor. Thanks! Removed in 69379adfcb65a3f1971a186c5be07988f41e721e